app/assets/js/rtm-upload-terms.js

```diff
@@ -76,28 +76,53 @@ if ( 'object' === typeof rtMediaHook ) {
             return args;
         }
 
+        var parent = false;
+        var isTermsEnabled = false;
+
+        var terms = $( '#rtmedia_upload_terms_conditions' );
         if ( 'uploader' === args.src ) {
-            if ( 'false' === rtmedia_upload_terms_data.uploader_terms_enabled ) {
-                return true;
-            }
 
-            var terms = $( '#rtmedia_upload_terms_conditions' );
             if ( terms.length === 0 ) {
-                rtp_display_terms_warning( $( '#drag-drop-area' ), rtmedia_upload_terms_data.message );
+                parent = $( '#drag-drop-area' );
+            } else {
+                parent = terms.parent( '.rtmedia-upload-terms' );
+            }
 
-                return false;
+            if ( 'true' === rtmedia_upload_terms_data.uploader_terms_enabled ) {
+                isTermsEnabled = true;
             }
+        } else if ( 'activity' === args.src ) {
 
-            rtmediaTermsConditionsElement = terms;
-            if ( terms.prop( 'checked' ) ) {
-                return true;
+            if ( terms.length === 0 ) {
+                parent = $( '#whats-new-options' );
             } else {
-                rtp_display_terms_warning( terms.parent( '.rtmedia-upload-terms' ), rtmedia_upload_terms_data.message );
+                parent = terms.parent( '.rtmedia-upload-terms' );
+            }
 
-                return false;
+            if ( 'true' === rtmedia_upload_terms_data.activity_terms_enabled ) {
+                isTermsEnabled = true;
             }
         }
 
+        if ( ! isTermsEnabled ) {
+            return args;
+        }
+
+        if ( terms.length === 0 ) {
+            rtp_display_terms_warning( parent, rtmedia_upload_terms_data.message );
+
```

Comment:
```suggestion
```

Reply:
Don't we leave blank line before `return`?

Reply:
Need but not in just two-line `if` block condition.

Reply:
Okay